include/xtensor/xmath.hpp

```diff
@@ -2307,6 +2346,42 @@ namespace detail {
             }
         };
 
+        struct nan_min
+        {
+            template <class T, class U>
+            constexpr auto operator()(const T lhs, const U rhs) const
+                -> std::common_type_t<T, U>
+            {
+                if (math::isnan(lhs))
+                {
+                    return rhs;
+                }
+                if (math::isnan(rhs))
+                {
+                    return lhs;
+                }
+                return detail::make_xfunction<math::minimum<void>>(lhs, rhs);
```

Comment:
The following should work on gcc 4.9, although I admit it is far less readable:

```cpp
return mat::isnan(lhs) ?
       rhs : 
       (math::isnan(rhs) ? lhs : detail::make_xfunction<math::minimum<void>>(lhs, rhs));
```

Reply:
Thanks! It works after adding a cast to the last term.

So gcc 4.9 treats `constexpr` using the rule in C++11: only one return statement is allowed.

Reply:
Yes, it's a bug that was fixed in gcc 5.